### DISCUSSION TITLE
Generating empty tags in another way

### DISCUSSION THREAD

**Discussion by karim-tao** [spatie/array-to-xml#202] [Q&A] [answered]
Hello everybody,
I need to generate xml files with empty elements in another way. I'll show you an example:

```
(new Spatie\ArrayToXml\ArrayToXml([
     'valore1' => 'ciao',
     'valore2' => null
]))->prettify()->toXml();
```

Current result:
```
<?xml version="1.0"?>
<root>
  <valore1>ciao</valore1>
  <valore2></valore2>
</root>
```

Result I would like:
```
<?xml version="1.0"?>
<root>
  <valore1>ciao</valore1>
  <valore2 />
</root>
```

It is possible to achieve this result with this library,
thank you all in advance

**Comment by weirdan** [comment, marked as answer]
Passing an empty array for that element should do the trick:

```php
(new Spatie\ArrayToXml\ArrayToXml([
     'valore1' => 'ciao',
     'valore2' => [], // <===============
]))->prettify()->toXml();
```

**Reply by karim-tao** [reply]
Thanks for the reply,
it might be interesting to implement an option to generate the empty tag using `null`.
